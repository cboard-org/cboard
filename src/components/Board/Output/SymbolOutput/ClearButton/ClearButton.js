import React, { Component } from 'react';
import PropTypes from 'prop-types';
import IconButton from '@material-ui/core/IconButton';
import ClearIcon from '@material-ui/icons/Clear';
import { Scannable } from 'react-scannable';

export class ClearButton extends Component {
  static propTypes = {
    /**
     * @ignore
     */
    hidden: PropTypes.bool
  };

  render() {
    const { hidden, increaseOutputButtons, ...other } = this.props;

    return (
      <Scannable disabled={hidden}>
        <IconButton
          aria-label="Clear"
          className={
            increaseOutputButtons ? 'Output__button__lg' : 'Output__button__sm'
          }
          {...other}
        >
          <ClearIcon
            className={
              increaseOutputButtons ? 'Output__icon__lg' : 'Output__icon__sm'
            }
          />
        </IconButton>
      </Scannable>
    );
  }
}

export default ClearButton;
