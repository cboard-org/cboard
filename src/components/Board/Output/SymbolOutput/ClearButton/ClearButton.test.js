import React from 'react';
import { shallowMatchSnapshot } from '../../../../../common/test_utils';

import ClearButton from './ClearButton';

describe('ClearButton tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<ClearButton />);
  });
});
