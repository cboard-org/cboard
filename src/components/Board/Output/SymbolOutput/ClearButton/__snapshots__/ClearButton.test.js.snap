// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearButton tests default renderer 1`] = `
<Unknown>
  <ForwardRef(WithStyles(ForwardRef(IconButton)))
    aria-label="Clear"
    className="Output__button__sm"
  >
    <UNDEFINED
      className="Output__icon__sm"
    />
  </ForwardRef(WithStyles(ForwardRef(IconButton)))>
</Unknown>
`;
